src/services/createResolvedRouteQuery.ts

```diff
@@ -1,10 +1,30 @@
 import { ResolvedRouteQuery } from '@/types/resolvedQuery'
 
+/**
+ * Creates a dumb wrapper around URLSearchParams because URLSearchParams cannot be correctly be proxied to support writing params to the RouterRoute
+ */
 export function createResolvedRouteQuery(query?: string): ResolvedRouteQuery {
   const params = new URLSearchParams(query)
 
   return {
-    get: (key: string) => params.get(key),
-    getAll: (key: string) => params.getAll(key),
+    get: (...args) => params.get(...args),
+    getAll: (...args) => params.getAll(...args),
+    set: (...args) => {
+      params.set(...args)
+    },
+    append: (...args) => {
+      params.append(...args)
+    },
+    delete: (...args) => {
+      params.delete(...args)
+    },
+    toString: (...args) => params.toString(...args),
+    forEach: (...args) => {
+      params.forEach(...args)
+    },
+    entries: (...args) => params.entries(...args),
+    keys: (...args) => params.keys(...args),
+    values: (...args) => params.values(...args),
+    has: (...args) => params.has(...args),
```

Comment:
wouldn't this also work?

```suggestion
    get: params.get,
    getAll: params.getAll,
    set: params.set,
    append: params.append,
    delete: params.delete,
    toString: params.toString,
    forEach: params.forEach,
    entries: params.entries,
    keys: params.keys,
    values: params.values,
    has: params.has,
```

Reply:
We tried this but the unit tests failed. Pretty sure its because this basically destructures the URLSearchParams and loses the context.